### PR TITLE
[ADO] secrets check in VarGrp for release & build

### PR DESCRIPTION
**Pull request by Garima-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1 [hunk 1]
```diff
@@ -120,31 +121,88 @@ class Release: ADOSVTBase
...
+                                if((($this.ReleaseObj[0].environments[$i].variableGroups) | Measure-Object).Count -gt 0)
+                                {
+                                    $varGrps += $this.ReleaseObj[0].environments[$i].variableGroups
+                                }
+                            }
+
+                            $varGrpObj = @();
+                            $varGrps | ForEach-Object {
+                                $varGrpURL = ("https://{0}.visualstudio.com/{1}/_apis/distributedtask/variablegroups/{2}") -f $($this.SubscriptionContext.SubscriptionName), $this.ProjectId, $_;
+                                $varGrpObj += [WebRequestHelper]::InvokeGetWebRequest($varGrpURL);
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1:153]: add try-catch

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1 [hunk 2]
```diff
@@ -120,31 +121,88 @@ class Release: ADOSVTBase
...
+                        $varGrps += $this.ReleaseObj[0].variableGroups
+                        if ((($this.ReleaseObj[0].environments) | Measure-Object).Count -gt 0) 
                         {
-                            $varList = $varList | select -Unique
-                            $controlResult.AddMessage([VerificationResult]::Failed, "Found secrets in release definition. Variables name: $varList" );
-                            $controlResult.SetStateData("List of variable name containing secret: ", $varList);
-                        }
-                        else {
-                            $controlResult.AddMessage([VerificationResult]::Passed, "No credentials found in release definition.");
+                            $envCount = ($this.ReleaseObj[0].environments).Count
```
Comment (abhaydaga) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1:139]: $envCount was already computed in the if clause. We can first calculate $envCount and then use it in if-clause. 10K builds -> 10K recomputations.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1 (hunk 1 above)
Comment (Garima-msft) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1]: done

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1 (hunk 2 above)
Comment (Garima-msft) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1]: done

**Review by abhaydaga — APPROVED** [review]

**Pull request merged**